### PR TITLE
prefer named-type rather than void-type to reduce typos

### PR DESCRIPTION
**Pull request by freemine**
in src/inc/taos.h, TAOS_xxx are typedef'd as void or void*.

by typedef'ing then with original type-name, not only benefit code-reviewing, but also reducing typos.

doing so, a few typos and bugs were found in current project, mostly, misusing TAOS where TAOS_RES is expected.

### CODE REVIEW

File: src/inc/taos.h [hunk 1]
```diff
@@ -22,12 +22,13 @@
...
-typedef void    TAOS_SUB;
-typedef void    TAOS_STREAM;
-typedef void    TAOS_STMT;
+typedef struct STscObj         TAOS;
+typedef struct STscStmt        TAOS_STMT;
+typedef struct SSqlObj         TAOS_RES;
+typedef struct SSqlStream      TAOS_STREAM;
+typedef struct SSub            TAOS_SUB;
+typedef unsigned char**        TAOS_ROW;
+
```
Comment (hjxilinx) [inline src/inc/taos.h:31]: We intend to hide the struct definition details from users. So, it seems that it is not a good idea to refactor code above.

**Review by hjxilinx — COMMENTED** [review]
@freemine

File: src/inc/taos.h (hunk 1 above)
Comment (freemine) [reply src/inc/taos.h:31]: typedef does not mean struct definition is open to user.

anyway, this is not a big issue, though it helps me find out the root cause of some runtime corruption.

**Review by hjxilinx — APPROVED** [review]

**Comment by sangshuduo** [comment]
0.00s$ mkdir debug
before_script.5
0.00s$ cd debug
1.35s$ cmake .. > /dev/null
The command "cmake .. > /dev/null" exited with 0.
16.26s$ make > /dev/null
/home/travis/build/taosdata/TDengine/src/client/src/tscProfile.c: In function ‘tscSaveSlowQueryFp’:
/home/travis/build/taosdata/TDengine/src/client/src/tscProfile.c:80:82: error: passing argument 8 of ‘taos_connect_a’ from incompatible pointer type [-Werror=incompatible-pointer-types]
   80 | (NULL, "monitor", tsInternalPass, "", 0, tscInitConnCb, sql, &tscSlowQueryConn);
      |                                                              ^~~~~~~~~~~~~~~~~
      |                                                              |
      |                                                              TAOS ** {aka struct STscObj **}
In file included from /home/travis/build/taosdata/TDengine/src/client/src/tscProfile.c:18:
/home/travis/build/taosdata/TDengine/src/client/inc/tsclient.h:450:7: note: expected ‘void **’ but argument is of type ‘TAOS **’ {aka ‘struct STscObj **’}
  450 | TAOS *taos_connect_a(char *ip, char *user, char *pass, char *db, uint16_t port, void (*fp)(void *, TAOS_RES *, int),
      |       ^~~~~~~~~~~~~~
cc1: all warnings being treated as errors
make[2]: *** [src/client/CMakeFiles/taos_static.dir/build.make:160: src/client/CMakeFiles/taos_static.dir/src/tscProfile.c.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:1503: src/client/CMakeFiles/taos_static.dir/all] Error 2
make: *** [Makefile:158: all] Error 2
The command "make > /dev/null" exited with 2.

**Review by guanshengliang — APPROVED** [review]

**Pull request merged**